src/NITTA/Model/Problems/Bind.hs

```diff
@@ -12,22 +12,44 @@ Stability   : experimental
 module NITTA.Model.Problems.Bind (
     Bind (..),
     BindProblem (..),
+    GroupBindingT (..),
 ) where
 
+import qualified Data.Set as S
 import Data.String.ToString
 import GHC.Generics
 import NITTA.Intermediate.Types
 
+data GroupBindingT = NonAlternativeBinds | AllBinds | FirstWaveBinds
+    deriving (Show, Generic, Eq, Ord)
+
 data Bind tag v x
     = Bind (F v x) tag
-    deriving (Generic, Eq)
+    | GroupBinding GroupBindingT [Bind tag v x]
+    deriving (Generic)
```

Comment:
Why do you make Bind and GroupBind one type? In this module, I don't see any differences, but in metrics, they should be different types.